seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSource.java

```diff
@@ -83,7 +88,7 @@ protected void buildSchemaWithConfig(Config pluginConfig) {
                     break;
                 default:
                     // TODO: use format SPI
-                    throw new UnsupportedOperationException("Unsupported format: " + format);
+                    throw new HttpConnectorException(HttpConnectorErrorCode.FORMAT_FAILED, new UnsupportedOperationException());
```

Comment:
Use `CommonErrorCode ILLEGAL_ ARGUMENT` may be better.

Reply:
ILLEGAL_ARGUMENT("COMMON-06", "Illegal argument") Illegal parameter exception, will there be ambiguity in this place, misleading users

Reply:
<img width="1288" alt="图片" src="https://user-images.githubusercontent.com/60029759/204437682-808bad26-98ed-4c8d-9bbd-429cc8e9ebdb.png">
This place will probably support more types in the future

Reply:
As for now, all connectors will use `CommonErrorCode.ILLEGAL_ARGUMENT` when handling this class exception, this exception can be understood as having incorrect parameters. So `CommonErrorCode.ILLEGAL_ARGUMENT` could be better. We can throw exception with clear error tips like the following shown:

```java
throw new HttpConnectorException(CommonErrorCode.ILLEGAL_ARGUMENT, String.format("Unsupported data format [%s], http connector only support json format now", format));
```

What do you think about it? @MonsterChenzhuo @FWLamb

Reply:
> As for now, all connectors will use `CommonErrorCode.ILLEGAL_ARGUMENT` when handling this class exception, this exception can be understood as having incorrect parameters. So `CommonErrorCode.ILLEGAL_ARGUMENT` could be better. We can throw exception with clear error tips like the following shown:
> 
> ```java
> throw new HttpConnectorException(CommonErrorCode.ILLEGAL_ARGUMENT, String.format("Unsupported data format [%s], http connector only support json format now", format));
> ```
> 
> What do you think about it? @MonsterChenzhuo @FWLamb

Agree with you.If more format types are supported later, we can point out in the error message so that users can know which formats are supported.

Reply:
I think it's reasonable.
@FWLamb @TyrantLucifer I have made the fix, please continue with the code review